src/main/flight/servos.c

```diff
@@ -326,49 +338,51 @@ void writeServos(void)
     case MIXER_CUSTOM_TRI:
         if (servosTricopterIsEnabledServoUnarmed()) {
             // if unarmed flag set, we always move servo
-            pwmWriteServo(servoIndex++, servo[SERVO_RUDDER]);
+            writeServoWithTracking(servoIndex++, SERVO_RUDDER);
         } else {
             // otherwise, only move servo when copter is armed
-            if (ARMING_FLAG(ARMED))
-                pwmWriteServo(servoIndex++, servo[SERVO_RUDDER]);
-            else
+            if (ARMING_FLAG(ARMED)) {
+                writeServoWithTracking(servoIndex++, SERVO_RUDDER);
+            } else {
                 pwmWriteServo(servoIndex++, 0); // kill servo signal completely.
+                servoWritten |= (1 << SERVO_RUDDER);
```

Comment:
This looks somewhat fishy and is hard to understand. Should we maybe set `servo[SERVO_RUDDER]` to 0 where it is calculated, and simplify here?